File: ngStore/Database/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ngStore.Database.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime OrderDate { get; set; }
        public string OrderNumber { get; set; }
        public int CustomerId { get; set; }
        public decimal TotalAmount { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; }
    }
}

File: ngStore/ngStoreClient/app/order/order.ts
import { OrderItem } from '../order/orderItem';
import { Customer } from '../customer/customer';
import { Product } from '../product/product';

export class Order {
    id: number;
    orderDate: Date = new Date();
    orderNumber: string;
    customer: Customer;
    totalAmount: number;
    subTotal: number;
    orderItems: Array<OrderItem> = new Array <OrderItem>();
    products: Array<Product> = new Array<Product>();
}